lib/lint.ml

```diff
@@ -295,7 +295,7 @@ module Lint = struct
 
   module Value = struct
     type t = {
-      is_macos : bool
+      host_is_macos : bool
```

Comment:
You could use a variant with the host here, instead of a boolean. That would ensure that code that needs to access the host information in the future can re-use this record.

Reply:
I've changed this to store the OS as a string instead, but storing the full variant would be a lot of work as it requires knowing how to find the distribution for every single OS as well as other information not yet necessary. (Dune has a lot of code to deal with this!)

Reply:
I'm not sure I see why a variant would be more complicated than a string. For instance, the `host_os` function can be written as:

```ocaml
let host_os =
  let os =
    Filename.quote_command "uname" ["-s"]
    |> Unix.open_process_in
    |> In_channel.input_line
    |> Option.value ~default:"Linux"
  in
  match String.lowercase_ascii os with
  | "darwin" | "osx" -> MacOS
  | s -> Other s
```

Reply:
Ah sorry I thought you meant the platform variants in `variant.ml` (which include lots more platform information), and not OCaml variants!